src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -217,6 +244,39 @@ private IReadOnlyList<SemanticRange> CombineSemanticRanges(params IReadOnlyList<
             return result;
         }
 
+        private async Task<SemanticTokens> GetMatchingCSharpResponseAsync(
+            TextDocumentIdentifier textDocumentIdentifier,
+            long? documentVersion,
+            CancellationToken cancellationToken)
+        {
+            var parameter = new SemanticTokensParams
+            {
+                TextDocument = textDocumentIdentifier,
+            };
+
+            ProvideSemanticTokensResponse csharpResponse;
+            for (var i = 0; i < 3; i++)
```

Comment:
Nit; extract the 3 out to a const, or annotate why 3.

Reply:
Because it's the loneliest number since the number 1? I usually default to 3 for retries just because it seems like you've given things a fair shake without wasting time, there's no logic to it beyond that.

Reply:
Why retry at all? Semantic tokens get polled multiple times per second. Would letting it fail and waiting for the next request be sufficient?

Reply:
I was trying to be robust, but it certainly simplifies things if I can kill the retry so I'm down.